jetty-core/jetty-util/src/main/java/org/eclipse/jetty/util/FileID.java

```diff
@@ -0,0 +1,380 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2022 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Locale;
+
+/**
+ * Simple, yet surprisingly common utility methods for identifying various file types commonly seen and worked with in a
+ * deployment scenario.
+ */
+public class FileID
+{
+    /**
+     * Does the provided path have a directory segment with
+     * the configured name.
+     *
+     * @param path the path to search
+     * @param directoryName the directory name to look for (case insensitive lookup)
+     * @return true if the directory name exists in path, false if otherwise
+     */
+    public static boolean containsDirectory(Path path, String directoryName)
+    {
+        if (path == null)
+            return false;
+        int segmentCount = path.getNameCount();
+        for (int i = segmentCount - 1; i >= 0; i--)
+        {
+            Path segment = path.getName(i);
+            if (segment.getFileName().toString().equalsIgnoreCase(directoryName))
+                return true;
+        }
+        return false;
+    }
+
+    /**
+     * Retrieve the basename of a path. This is the name of the
+     * last segment of the path, with any dot suffix (e.g. ".war") removed
+     *
+     * @param path The string path
+     * @return The last segment of the path without any dot suffix
+     */
+    public static String getBasename(Path path)
+    {
+        String basename = path.getFileName().toString();
```

Comment:
You should test for `path.getFileName()` returning `null`.

Reply:
Adding testcase